### PR TITLE
fix(ui): TE-2056 improve clarity of error messages

### PR DESCRIPTION
**Pull request by harshilvelotio**
#### Issue(s)

- https://startree.atlassian.net/browse/TE-2056

#### Description

- The goal of this PR is to improve error messaging across ThirdEye
- There are 3 levels of error messaging now
    - Error code mapped user-friendly error messages: here we have recreated the following map from the backend on the UI for more user-friendly error messaging based on different error codes https://github.com/startreedata/thirdeye/blob/6d893fed07a2283f734a4540d53ee90ae13f6c04/thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/ThirdEyeStatus.java#L18
    - Generic error messages: Instead of showing large not-so-user-friendly error message we will now show a more generic user friendly error message
    - The error message alert now has a "Show excpetions" section that is collapsed by default and on expanding shows the large backend sent exception for easier debugging
    - For RBAC related errors we will now show a toast message "You are not authorised to perform this action" for better clarity
- Also fixed all unit tests across the app

#### Screenshots
image
image
image
image

#### How to test
- See if you can try to reproduce some error case scenarios and check out the error toast message text

List steps to reproduce the issue(s) and test the change, if required.

### CODE REVIEW

File: thirdeye-ui/src/app/utils/rest/rest.util.ts [hunk 1]
```diff
@@ -13,21 +13,195 @@
...
+    ERR_INVALID_PARAMS_COMPONENTS = "ERR_INVALID_PARAMS_COMPONENTS",
+    ERR_INVALID_DETECTION_REGRESSORS = "ERR_INVALID_DETECTION_REGRESSORS",
+    ERR_INVALID_SQL = "ERR_INVALID_SQL",
+    ERR_NEGATIVE_LIMIT_VALUE = "ERR_NEGATIVE_LIMIT_VALUE",
+    ERR_NEGATIVE_OFFSET_VALUE = "ERR_NEGATIVE_OFFSET_VALUE",
+    ERR_OFFSET_WITHOUT_LIMIT = "ERR_OFFSET_WITHOUT_LIMIT",
+}
+
+export const getErrorMessages = (error: AxiosError): ErrorMessage[] => {
+    const errorCodeMessageMap: Record<ErrorCode, string> = {
```
Comment (alcatraz627) [inline thirdeye-ui/src/app/utils/rest/rest.util.ts:65]: Nitpick: 
Objects without any dependence on the function params or state can be declared outside the function for a slightly less memory overload

**Review by alcatraz627 — APPROVED** [review]

**Review by cyrilou242 — CHANGES_REQUESTED** [review]
Let's change the toggle from 
"Show Exception"
to 
"Show details"

then LGTM

**Review by cyrilou242 — APPROVED** [review]
LGTM! great work

**Comment by cyrilou242** [comment]
merging because I'm cutting a release right now

**Pull request merged**